bears/python/PEP8NotebookBear.py

```diff
@@ -0,0 +1,130 @@
+import autopep8
+from copy import copy
+import nbformat
+
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from coalib.bears.LocalBear import LocalBear
+from coalib.bears.requirements.PipRequirement import PipRequirement
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.settings.Setting import typed_list
+
+
+class NotebookHandler:
+    """Utility class for Jupyter Notebook handling."""
+
+    def __init__(self, filename, file):
+        """
+        Constructs a notebook handler to deal with NotebookNode objects.
+        A NotebookNode is a dict-like node with attribute-access.
+
+        :param filename: The filename of the file.
+        :param file:     The file contents as string array.
+        """
+        self.filename = filename
+        self.file = file
+        self._file_nb = self.read_from_list(file)
+
+    @staticmethod
+    def read_from_list(file):
+        """
+        Reads a notebook from a string array and returns the NotebookNode
+        object.
+
+        :param file: The notebook file contents as JSON string array.
+        :return:     The notebook as NotebookNode.
+        """
+        return nbformat.reads(''.join(file), nbformat.NO_CONVERT)
+
+    def write_to_list(self, file_nb):
```

Comment:
I think to avoid confusion, it would be a good idea to add `newline__at_eof` or so. If it is True/False, use that.
If it is `None`, use `self.file[-1].endswith('\n')`

Also, a more descriptive docstring mentioning it auto-understands the EOF \n